awsauth_test.go

```diff
@@ -420,25 +418,27 @@ func TestAWSParseAccountIDAndPartitionFromARN(t *testing.T) {
 	}
 }
 
-func TestAWSGetCredentials_shouldError(t *testing.T) {
+func TestAWSGetCredentials_shouldErrorWhenBlank(t *testing.T) {
 	resetEnv := unsetEnv(t)
 	defer resetEnv()
-	cfg := Config{}
 
+	cfg := Config{}
 	c, err := GetCredentials(&cfg)
-	if awsErr, ok := err.(awserr.Error); ok {
-		if awsErr.Code() != "NoCredentialProviders" {
-			t.Fatal("Expected NoCredentialProviders error")
+
+	if err == nil {
```

Comment:
The logic within this branch seems like the normal flow of execution so I would be inclined to breaking it out of the if/else condition and instead branch if `err != nil`. In Go, you will find that else statements are not used that often because the community tends to lean more towards only using the if statement as guard clauses for conditions that would break the normal flow of execution. For example, if `err != nil` then exit the test. 

Here's a blog post that may helpful better explain this comment https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88